addon/mixins/form-data-adapter.js

```diff
@@ -13,13 +13,16 @@ export default Ember.Mixin.create({
 
     if (typeof FormData !== 'undefined' && data && this.formDataTypes.contains(type)) {
       var formData = new FormData();
-      var root = Ember.keys(data)[0];
-
-      Ember.keys(data[root]).forEach(function(key) {
-        if (typeof data[root][key] !== 'undefined') {
-          formData.append(root + "[" + key + "]", data[root][key]);
+      var fields = this.getDataToTransform(data);
+
+      Ember.keys(fields).forEach(function(key) {
+        if (typeof fields[key] !== 'undefined') {
+          formData.append(
+            this.getFormKey(key, fields),
+            this.getFormValue(key, fields)
```

Comment:
As in `.forEach()` I would pass the collection you're iterating over, you could be depending of relationships among the whole data being serialized. Anyway, as I can not imagine a clear use case for this right now, I agree on passing only `key, fields[key]`.